src/openassetio-python/CMakeLists.txt

```diff
@@ -5,3 +5,9 @@
 add_subdirectory(bridge)
 # Python C extension module making C++ types usable in Python.
 add_subdirectory(module)
+
+# Install OpenAssetIO pure Python into the venv.
+install(
+    DIRECTORY ${PROJECT_SOURCE_DIR}/python/openassetio
+    DESTINATION "${OPENASSETIO_PYTHON_SITEDIR}"
+)
```

Comment:
[question] Do we need to be careful to only copy `.py` here? Can we filter it?

Reply:
We can use `FILES_MATCHING PATTERN`, but is that overkill?  Currently, at least, I think we want everything.

Reply:
Is there some danger of accidentally including `.pyc` or other artifacts from developer tooling? We used to end up with `egg-info` and such as well.  I guess the same argument as the fact we name includes/cpp vs glbobing in CMake?

Reply:
Also, when we point `setup.py` to that dir, iirc, it will _only_ include `.py` without reconfiguration, so a setup.py assembled package could be different to the cmake assembled one?

Reply:
Addressed in 0d16ac2784f441217323b2ecf7b1be993fe7851f